packages/core/src/components/fields/LayoutGridField.tsx

```diff
@@ -140,6 +139,594 @@ function isNumericIndex(str: string) {
+      schema={schema}
+      layoutGridSchema={layoutGridSchema}
+    />
+  ));
+}
+
+/**
+ * The props for the LayoutGridFieldComponent.
+ */
+type LayoutGridFieldComponentProps<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any,
+> = LayoutGridFieldProps<T, S, F> & {
+  /** The string or object that represents the configuration for the grid field */
+  gridSchema?: ConfigObject | string;
+};
+
+/** Renders the field described by `gridSchema`. If `gridSchema` is not an object, then is will be assumed
+ * to be the dotted-path to the field in the schema. Otherwise, we extract the `name`, and optional `render` and all
+ * other props. If `name` does not exist and there is an optional `render`, we return the `render` component with only
+ * specified props for that component. If `name` exists, we take the name, the initial & root schemas and the formData
+ * and get the destination schema, is required state and optional oneOf/anyOf options for it. If the destination
+ * schema was located along with oneOf/anyOf options then a `LayoutMultiSchemaField` will be rendered with the
+ * `uiSchema`, `errorSchema`, `fieldPathId` and `formData` drilled down to the dotted-path field, spreading any other
+ * props from `gridSchema` into the `ui:options`. If the destination schema located without any oneOf/anyOf options,
+ * then a `SchemaField` will be rendered with the same props as mentioned in the previous sentence. If no destination
+ * schema was located, but a custom render component was found, then it will be rendered with many of the non-event
+ * handling props. If none of the previous render paths are valid, then a null is returned.
+ *
+ * @returns - One of `LayoutMultiSchemaField`, `SchemaField`, a custom render component or null, depending
+ */
+function LayoutGridFieldComponent<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
+  props: LayoutGridFieldComponentProps<T, S, F>,
+) {
+  const { gridSchema, ...layoutGridFieldProps } = props;
+  const {
+    schema: initialSchema,
+    uiSchema,
+    errorSchema,
+    fieldPathId,
+    onBlur,
+    onFocus,
+    formData,
+    readonly,
+    registry,
+    layoutGridSchema, // Used to pull this out of otherProps since we don't want to pass it through
+    ...otherProps
+  } = layoutGridFieldProps;
+  const { onChange } = otherProps;
+  const { fields } = registry;
+  const { SchemaField, LayoutMultiSchemaField } = fields;
+
+  const uiComponentProps = computeUIComponentPropsFromGridSchema(registry, gridSchema);
+  const { name, UIComponent, uiProps } = uiComponentProps;
+  const {
+    schema,
+    isRequired,
+    isReadonly,
+    optionsInfo,
+    fieldPathId: fieldIdSchema,
+  } = getSchemaDetailsForField<T, S, F>(registry, name, initialSchema, formData, fieldPathId);
+  const memoFieldPathId = useDeepCompareMemo<FieldPathId>(fieldIdSchema);
+
+  if (uiComponentProps.rendered) {
```

Comment:
It kind of sucks that the rules of hooks forces us to call that rather expensive `getSchemaDetailsForField()` first